tensorflow/compiler/mlir/tosa/tests/tf-to-tosa-pipeline.mlir

```diff
@@ -764,23 +771,103 @@ func @test_depth_to_space(%arg0: tensor<1x32x32x8xf32>) -> tensor<1x64x64x2xf32>
+// CHECK: tosa.arithmetic_right_shift
+func @test_right_shift(%arg0: tensor<4x4xi32>, %arg1: tensor<1x1xi32>) -> tensor<4x4xi32> {
+  %0 = "tf.RightShift"(%arg0, %arg1) : (tensor<4x4xi32>, tensor<1x1xi32>) -> tensor<4x4xi32>
+  return %0 : tensor<4x4xi32>
+}
+
+// -----
+
+// CHECK-LABEL: test_one_hot
+// CHECK: %[[VAR0:.*]] = "tosa.const"() {value = dense<[0, 2, 1]> : tensor<3xi32>} : () -> tensor<3xi32>
+// CHECK: %[[VAR1:.*]] = "tosa.reshape"(%arg1) {new_shape = [1, 1, 1]} : (tensor<f32>) -> tensor<1x1x1xf32>
+// CHECK: %[[VAR2:.*]] = "tosa.tile"(%[[VAR1]]) {multiples = [16, 1, 1]} : (tensor<1x1x1xf32>) -> tensor<16x1x1xf32>
+// CHECK: %[[VAR3:.*]] = "tosa.reshape"(%arg2) {new_shape = [1, 1, 1]} : (tensor<f32>) -> tensor<1x1x1xf32>
+// CHECK: %[[VAR4:.*]] = "tosa.tile"(%[[VAR3]]) {multiples = [16, 2, 1]} : (tensor<1x1x1xf32>) -> tensor<16x2x1xf32>
+// CHECK: %[[VAR5:.*]] = "tosa.reshape"(%arg0) {new_shape = [16, 1]} : (tensor<4x4xi32>) -> tensor<16x1xi32>
+// CHECK: %[[VAR6:.*]] = "tosa.scatter"(%[[VAR4]], %[[VAR5]], %[[VAR2]]) : (tensor<16x2x1xf32>, tensor<16x1xi32>, tensor<16x1x1xf32>) -> tensor<16x2x1xf32>
+// CHECK: %[[VAR7:.*]] = "tosa.reshape"(%[[VAR6]]) {new_shape = [16, 1, 2]} : (tensor<16x2x1xf32>) -> tensor<16x1x2xf32>
+// CHECK: %[[VAR8:.*]] = "tosa.transpose"(%[[VAR7]], %[[VAR0]]) : (tensor<16x1x2xf32>, tensor<3xi32>) -> tensor<16x2x1xf32>
+// CHECK: %[[VAR9:.*]] = "tosa.reshape"(%[[VAR8]]) {new_shape = [4, 4, 2]} : (tensor<16x2x1xf32>) -> tensor<4x4x2xf32>
+func @test_one_hot(%arg0: tensor<4x4xi32>, %arg1: tensor<f32>, %arg2: tensor<f32>) -> tensor<4x4x2xf32> {
+  %0 = "tf.Const"()  {value = dense<2> : tensor<i32>}  : () -> tensor<i32>
+  %1 = "tf.OneHot"(%arg0, %0, %arg1, %arg2) {axis = -1 : i64} : (tensor<4x4xi32>, tensor<i32>, tensor<f32>, tensor<f32>) -> tensor<4x4x2xf32>
+  return %1 : tensor<4x4x2xf32>
+}
+
+// -----
+
 // CHECK-LABEL: test_fakequant_with_min_max_args
-// CHECK-DAG: "tosa.const"() {value = dense<16383.75> : tensor<f32>}
-// CHECK-DAG: "tosa.const"() {value = dense<-1.000000e+00> : tensor<f32>}
+// CHECK-DAG: "tosa.const"() {value = dense<-2.00003052> : tensor<f32>} 
+// CHECK-DAG: "tosa.const"() {value = dense<1.99996948> : tensor<f32>}
 // CHECK-DAG: "tosa.const"() {value = dense<6.10360876E-5> : tensor<f32>}
+// CHECK-DAG: "tosa.const"() {value = dense<16383.75> : tensor<f32>}
+// CHECK-DAG: "tosa.const"() {value = dense<5.000000e-01> : tensor<f32>}
 // CHECK: tosa.reshape
-// CHECK: tosa.mul
+// CHECK: tosa.minimum
 // CHECK: tosa.reshape
-// CHECK: tosa.add
-// CHECK: tosa.cast
-// CHECK: tosa.rescale
-// CHECK: tosa.rescale
-// CHECK: tosa.cast
+// CHECK: tosa.maximum
 // CHECK: tosa.reshape
 // CHECK: tosa.sub
 // CHECK: tosa.reshape
 // CHECK: tosa.mul
+// CHECK: tosa.reshape
+// CHECK: tosa.add
+// CHECK: tosa.floor
+// CHECK: tosa.reshape
+// CHECK: tosa.mul
+// CHECK: tosa.reshape
+// CHECK: tosa.add
 func @test_fakequant_with_min_max_args(%arg0: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
```

Comment:
Perform the same value validation on the rest of the added tests. If they are short (e.g. less than 3-4 ops) its fine without them but most of these tests are pretty complex and we want to avoid future errors.

Reply:
The plan is to clean up all the tests in a separate PR. Is that a workable option ? We tried the test_one_hot to get feedback on the right way to go about it, rather than change everything, then potentially have to do a second pass at everything.